mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpsTypes.h

```diff
@@ -439,6 +465,23 @@ class AccessorImplDeviceType
   llvm::ArrayRef<mlir::Type> getBody() const;
 };
 
+class AccessorSubscriptType
+    : public Type::TypeBase<AccessorSubscriptType, Type,
+                            detail::AccessorSubscriptStorage,
+                            mlir::MemRefElementTypeInterface::Trait,
+                            mlir::LLVM::PointerElementTypeInterface::Trait> {
+public:
+  using Base::Base;
+
+  static mlir::sycl::AccessorSubscriptType
+  get(MLIRContext *Context, unsigned int CurrentDimension,
+      llvm::SmallVector<mlir::Type, 4> Body);
+  static mlir::Type parseType(mlir::DialectAsmParser &Parser);
+
+  unsigned int getCurrentDimension() const;
```

Comment:
Should we have an additional `mlir::AccessorType` member here hosting the type of the underlying accessor? This would allow checking this type is the correct output of our `sycl.accessor.subindex_operator` operation.

Reply:
I'd rather have it accessible using a function (to perform the checks more easily).

Reply:
I added `getAccessorType()` for easy accessibility.